docs/creating-new/custom.rst

```diff
@@ -4,6 +4,179 @@
+
+    # Check variables is not empty
+    hunter_test_string_not_empty("@HUNTER_SELF@")
+    hunter_test_string_not_empty("@HUNTER_EP_NAME@")
+    hunter_test_string_not_empty("@HUNTER_PACKAGE_URL@")
+    hunter_test_string_not_empty("@HUNTER_PACKAGE_SHA1@")
+    hunter_test_string_not_empty("@HUNTER_PACKAGE_DOWNLOAD_DIR@")
+    hunter_test_string_not_empty("@HUNTER_PACKAGE_SOURCE_DIR@")
+    hunter_test_string_not_empty("@HUNTER_INSTALL_PREFIX@")
+
+    ExternalProject_Add(
+        @HUNTER_EP_NAME@ # Name of the external project. Actually not used set for beautify logging messages
+        URL
+        @HUNTER_PACKAGE_URL@ # URL of the package to download
+        URL_HASH
+        SHA1=@HUNTER_PACKAGE_SHA1@ # SHA1 hash
+        DOWNLOAD_DIR
+        "@HUNTER_PACKAGE_DOWNLOAD_DIR@" # Archive destination location
+        SOURCE_DIR
+        "@HUNTER_PACKAGE_SOURCE_DIR@" # Unpack directory
+        INSTALL_DIR
+        "@HUNTER_INSTALL_PREFIX@" # not used actually (see install command)
+        CONFIGURE_COMMAND
+        ""
+        BUILD_COMMAND
+        xcodebuild -target ios-sim -configuration Release
+        BUILD_IN_SOURCE
+        1
+        INSTALL_COMMAND
+        "@CMAKE_COMMAND@" -E copy build/Release/ios-sim "@HUNTER_INSTALL_PREFIX@"
+    )
+
+Next steps are similar to :ref:`cmake-simple`.
+
+.. code-block:: bash
+
+    > cat cmake/projects/ios_sim/hunter.cmake
+
+    # !!! DO NOT PLACE HEADER GUARDS HERE !!!
+
+    include(hunter_add_version)
+    include(hunter_download)
+    include(hunter_pick_scheme)
+
+    hunter_add_version(
+        PACKAGE_NAME
+        ios_sim
+        VERSION
+        "1.8.2"
+        URL
+        "https://github.com/phonegap/ios-sim/archive/1.8.2.tar.gz"
+        SHA1
+        4328b3c8e6b455631d52b7ce5968170c9769eb1e
+    )
+
+    hunter_pick_scheme(DEFAULT url_sha1_ios_sim) # Use new custom scheme
+    hunter_download(PACKAGE_NAME ios_sim)
+
+.. code-block:: bash
+
+    > grep ios_sim cmake/config/default.cmake
+    hunter_config(ios_sim VERSION 1.8.2)
+
+#. Using
+Now package ready to be used:
+
+.. code-block:: bash
+
+    > cat CMakeLists.txt
+    cmake_minimum_required(VERSION 2.8)
+
+    include("cmake/HunterGate.cmake")
+
+    HunterGate(
+        URL "https://url/to/your/hunter-archive.tar.gz"
+        SHA1 "put-archive-sha1-here"
+    )
+
+    hunter_add_package(ios_sim)
+
+    find_program(IOS_SIM_EXECUTABLE ios-sim ${IOS_SIM_ROOT})
+    message("ios_sim: ${IOS_SIM_EXECUTABLE}")
+    > cmake -H. -B_builds
+    -- [hunter] HUNTER_ROOT: /.../Hunter
+    -- [hunter] [ Hunter-ID: 7912489 | Config-ID: 9ec2ff8 | Toolchain-ID: c018e63 ]
+    -- [hunter] IOS_SIM_ROOT: /.../Hunter/_Base/7912489/9ec2ff8/c018e63/Install (ver.: 1.8.2)
+    ...
+    -- downloading...
+         src='https://github.com/phonegap/ios-sim/archive/1.8.2.tar.gz'
+    -- [download 100% complete]
+    ios_sim: /.../Hunter/_Base/7912489/9ec2ff8/c018e63/Install/ios-sim
+
+### Default behaviour
```

Comment:
`### Something` is not RST format, should be:
```
Default behavior
================
```

Reply:
OK